### PR TITLE
add tests, CI, better docs

### PR DESCRIPTION
**Pull request by vincentsarago**
this PR does: 

- add tests 
- add method to choose between jpg or png (if no extension provided) 
- add circle-ci
- add codecov
- use docker-compose 
- update API doc
- add pre-commit
- add tox

### CODE REVIEW

File: lambda_tiler/handler.py [hunk 1]
```diff
@@ -77,19 +78,24 @@ class TilerError(Exception):
...
+    host = event["headers"].get("X-Forwarded-Host", event["headers"].get("Host", ""))
+    # Check for API gateway stage
+    if ".execute-api." in host and ".amazonaws.com" in host:
+        stage = event["requestContext"].get("stage", "")
+        host = f"{host}/{stage}"
+
+    scheme = "http://" if host.startswith("127.0.0.1") else "https://"
+    endpoint = f"{scheme}{host}"
+
+    qs = [f"{k}={v}" for k, v in kwargs.items()]
```
Comment (vincentsarago) [inline lambda_tiler/handler.py:93]: @bitner I've change the way we determine the endpoint URL. could you double check that it's still working for you ?

File: lambda_tiler/handler.py [hunk 2]
```diff
@@ -162,64 +260,69 @@ def metadata(
...
+        )
     else:
-        tile, mask = main.tile(
-            url, x, y, z, indexes=indexes, tilesize=tilesize)
+        tile, mask = cogTiler.tile(
+            url, x, y, z, indexes=indexes, tilesize=tilesize, nodata=nodata
+        )
+
+    if not ext:
+        ext = "jpg" if mask.all() else "png"
```
Comment (vincentsarago) [inline lambda_tiler/handler.py:300]: @palmerj I think this should fix what we discussed. basically if no `ext` is given (`/tiles/1/2/3`) the code will choose between jpg or png

File: lambda_tiler/handler.py (hunk 1 above)
Comment (bitner) [reply lambda_tiler/handler.py:93]: I won't be able to check today. I'll update and check tomorrow. It looks like should still be fine though.

Comment (bitner) [reply lambda_tiler/handler.py:93]: didn't get to checking yesterday. if you break it, i'll pr a fix :-)

**Pull request merged**